olive/cache.py

```diff
@@ -109,10 +110,49 @@ def clean_pass_run_cache(pass_type: str, cache_dir: Union[str, Path] = ".olive-c
         _delete_run(run_json.stem, cache_dir)
 
 
+def get_non_local_resource(resource_path: ResourcePath, cache_dir: Union[str, Path] = ".olive-cache"):
```

Comment:
add unit test for the get_non_local_resource?

Reply:
Done